src/modules/tensor/hip/kernel/snow.cpp

```diff
@@ -0,0 +1,459 @@
+        snow_hip_compute(srcPtr, &pix_f8, brightCoeff, snowThresh, dark);
+        rpp_hip_pack_float8_and_store8(dstPtr + dstIdx, &pix_f8);
+    }
+}
+
+template <typename T>
+__global__ void snow_pkd3_pln3_hip_tensor(T *srcPtr,
+                                          uint2 srcStridesNH,
+                                          T *dstPtr,
+                                          uint3 dstStridesNCH,
+                                          float *brightnessCoefficient,
+                                          float *snowThreshold,
+                                          int *darkMode,
+                                          RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        return;
+
+    uint srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + ((id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3);
+    uint dstIdx = (id_z * dstStridesNCH.x) + (id_y * dstStridesNCH.z) + id_x;
+
+    float brightCoeff = brightnessCoefficient[id_z];
+    float snowThresh = ((snowThreshold[id_z] * 127.5f) + 85.0f) * ONE_OVER_255;
+    int dark = darkMode[id_z];
+
+    d_float24 pix_f24;
+
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &pix_f24);
+    snow_hip_compute(srcPtr, &pix_f24, brightCoeff, snowThresh, dark);
+    rpp_hip_pack_float24_pln3_and_store24_pln3(dstPtr + dstIdx, dstStridesNCH.y, &pix_f24);
+}
+
+template <typename T>
+__global__ void snow_pln3_pkd3_hip_tensor(T *srcPtr,
+                                          uint3 srcStridesNCH,
+                                          T *dstPtr,
+                                          uint2 dstStridesNH,
+                                          float *brightnessCoefficient,
+                                          float *snowThreshold,
+                                          int *darkMode,
+                                          RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        return;
+
+    uint srcIdx = (id_z * srcStridesNCH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    float brightCoeff = brightnessCoefficient[id_z];
+    float snowThresh = ((snowThreshold[id_z] * 127.5f) + 85.0f) * ONE_OVER_255;
```

Comment:
We should absorb the ONE_OVER_255 into the 127.5 and the 85.
Then this whole thing becomes 1 fmaf instruction instead of 2 mul + 1 add